### ISSUE TITLE
Error inflating class com.instabug.library.view.IconView

### ISSUE THREAD

**Issue by dimmetrius** [Instabug/Instabug-Android#198]
### Steps to Reproduce the Problem
trying to add instabug to simple flutter application

add to build.gradle
```gradle
implementation 'com.instabug.library:instabug:8.0.3'
```

created a new Java class that extends FlutterApplication and add it to AndroidManifest.xml.

```java
public class CustomFlutterApplication extends FlutterApplication {
    @Override
    public void onCreate() {
        super.onCreate();
        new Instabug.Builder(CustomFlutterApplication.this, "XXXXXX")
                .setInvocationEvents(InstabugInvocationEvent.SHAKE, InstabugInvocationEvent.SCREENSHOT)
                .build();
        Instabug.setWelcomeMessageState(WelcomeMessage.State.DISABLED);

        CrashReporting.setState(Feature.State.DISABLED);
        BugReporting.setReportTypes(BugReporting.ReportType.BUG, BugReporting.ReportType.FEEDBACK);
    }
}
```
### Actual Behavior
crash on opening feedback screen

```java
Instabug.show();
```
E/AndroidRuntime( 5263): android.view.InflateException: Binary XML file line #60: Binary XML file line #60: Error inflating class com.instabug.library.view.IconView
E/AndroidRuntime( 5263): Caused by: android.view.InflateException: Binary XML file line #60: Error inflating class com.instabug.library.view.IconView
E/AndroidRuntime( 5263): Caused by: java.lang.reflect.InvocationTargetException
E/AndroidRuntime( 5263): 	at java.lang.reflect.Constructor.newInstance0(Native Method)
E/AndroidRuntime( 5263): 	at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
E/AndroidRuntime( 5263): 	at android.view.LayoutInflater.createView(LayoutInflater.java:647)
E/AndroidRuntime( 5263): 	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:790)
E/AndroidRuntime( 5263): 	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:730)
E/AndroidRuntime( 5263): 	at android.view.LayoutInflater.rInflate(LayoutInflater.java:863)
E/AndroidRuntime( 5263): 	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
E/AndroidRuntime( 5263): 	at android.view.LayoutInflater.rInflate(LayoutInflater.java:866)
E/AndroidRuntime( 5263): 	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
E/AndroidRuntime( 5263): 	at android.view.LayoutInflater.inflate(LayoutInflater.java:515)
E/AndroidRuntime( 5263): 	at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
E/AndroidRuntime( 5263): 	at com.instabug.bug.view.a.onCreateViewHolder(AttachmentsAdapter.java:82)
E/AndroidRuntime( 5263): 	at androidx.recyclerview.widget.RecyclerView$Adapter.createViewHolder(RecyclerView.java:6794)
E/AndroidRuntime( 5263): 	at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:5975)
E/AndroidRuntime( 5263): 	at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5858)
E/AndroidRuntime( 5263): 	at androidx.recyclerview.widget.RecyclerView$Recycler.getViewForPosition(RecyclerView.java:5854)
E/AndroidRuntime( 5263): 	at androidx.recyclerview.widget.LinearLayoutManager$LayoutState.next(LinearLayoutManager.java:2230)
E/AndroidRuntime( 5263): 	at androidx.recyclerview.widget.LinearLayoutManager.layoutChunk(LinearLayoutManager.java:1557)
E/AndroidRuntime( 5263): 	at androidx.recyclerview.widget.LinearLayoutManager.fill(LinearLayoutManager.java:1517)
E/AndroidRuntime( 5263): 	at androidx.recyclerview.widget.LinearLayoutManager.onLayoutChildren(LinearLayoutManager.java:612)
E/AndroidRuntime( 5263): 	at androidx.recyclerview.widget.RecyclerView.dispatchLayoutStep2(RecyclerView.java:3924)
E/AndroidRuntime( 5263): 	at androidx.recyclerview.widget.RecyclerView.dispatchLayout(RecyclerView.java:3641)
E/AndroidRuntime( 5263): 	at androidx.recyclerview.widget.RecyclerView.onLayout(RecyclerView.java:4194)
E/AndroidRuntime( 5263): 	at android.view.View.layout(View.java:20672)
E/AndroidRuntime( 5263): 	at android.view.ViewGroup.layout(ViewGroup.java:6194)
E/AndroidRuntime( 5263): 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1812)
E/AndroidRuntime( 5263): 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1656)
E/AndroidRuntime( 5263): 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1565)
E/AndroidRuntime( 5263): 	at android.view.View.layout(View.java:20672)
E/AndroidRuntime( 5263): 	at android.view.ViewGroup.layout(ViewGroup.java:6194)
E/AndroidRuntime( 5263): 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
E/AndroidRuntime( 5263): 	at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
E/AndroidRuntime( 5263): 	at android.widget.ScrollView.onLayout(ScrollView.java:1552)
E/AndroidRuntime( 5263): 	at android.view.View.layout(View.java:20672)
E/AndroidRuntime( 5263): 	at android.view.ViewGroup.layout(ViewGroup.java:6194)
E/AndroidRuntime( 5263): 	at androidx.coordinatorlayout.widget.CoordinatorLayout.layoutChild(CoordinatorLayout.java:1183)
E/AndroidRuntime( 5263): 	at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayoutChild(CoordinatorLayout.java:870)
E/AndroidRuntime( 5263): 	at androidx.coordinatorlayout.widget.CoordinatorLayout.onLayout(CoordinatorLayout.java:889)
E/AndroidRuntime( 5263): 	at android.view.View.layout(View.java:20672)
E/AndroidRuntime( 5263): 	at android.view.ViewGroup.layout(ViewGroup.java:6194)
E/AndroidRuntime( 5263): 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
E/AndroidRuntime( 5263): 	at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
E/AndroidRuntime( 5263): 	at android.view.View.layout(View.java:20672)
E/AndroidRuntime( 5263): 	at android.view.ViewGroup.layout(ViewGroup.java:6194)
E/AndroidRuntime( 5263): 	at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1083)
E/AndroidRuntime( 5263): 	at android.view.View.layout(View.java:20672)
E/AndroidRuntime( 5263): 	at android.view.ViewGroup.layout(ViewGroup.java:6194)
E/AndroidRuntime( 5263): 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1812)
E/AndroidRuntime( 5263): 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1656)
E/AndroidRuntime( 5263): 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1565)
E/AndroidRuntime( 5263): 	at android.view.View.layout(View.java:20672)
E/AndroidRuntime( 5263): 	at android.view.ViewGroup.layout(ViewGroup.java:6194)
E/AndroidRuntime( 5263): 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
E/AndroidRuntime( 5263): 	at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
E/AndroidRuntime( 5263): 	at android.view.View.layout(View.java:20672)
E/AndroidRuntime( 5263): 	at android.view.ViewGroup.layout(ViewGroup.java:6194)
E/AndroidRuntime( 5263): 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1812)
E/AndroidRuntime( 5263): 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1656)
E/AndroidRuntime( 5263): 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1565)
E/AndroidRuntime( 5263): 	at android.view.View.layout(View.java:20672)
E/AndroidRuntime( 5263): 	at android.view.ViewGroup.layout(ViewGroup.java:6194)
E/AndroidRuntime( 5263): 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
E/AndroidRuntime( 5263): 	at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
E/AndroidRuntime( 5263): 	at android.view.View.layout(View.java:20672)
E/AndroidRuntime( 5263): 	at android.view.ViewGroup.layout(ViewGroup.java:6194)
E/AndroidRuntime( 5263): 	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1812)
E/AndroidRuntime( 5263): 	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1656)
E/AndroidRuntime( 5263): 	at android.widget.LinearLayout.onLayout(LinearLayout.java:1565)
E/AndroidRuntime( 5263): 	at android.view.View.layout(View.java:20672)
E/AndroidRuntime( 5263): 	at android.view.ViewGroup.layout(ViewGroup.java:6194)
E/AndroidRuntime( 5263): 	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:323)
E/AndroidRuntime( 5263): 	at android.widget.FrameLayout.onLayout(FrameLayout.java:261)
E/AndroidRuntime( 5263): 	at com.android.internal.policy.DecorView.onLayout(DecorView.java:753)
E/AndroidRuntime( 5263): 	at android.view.View.layout(View.java:20672)
E/AndroidRuntime( 5263): 	at android.view.ViewGroup.layout(ViewGroup.java:6194)
E/AndroidRuntime( 5263): 	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2792)
E/AndroidRuntime( 5263): 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2319)
E/AndroidRuntime( 5263): 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1460)
E/AndroidRuntime( 5263): 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:7183)
E/AndroidRuntime( 5263): 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:949)
E/AndroidRuntime( 5263): 	at android.view.Choreographer.doCallbacks(Choreographer.java:761)
E/AndroidRuntime( 5263): 	at android.view.Choreographer.doFrame(Choreographer.java:696)
E/AndroidRuntime( 5263): 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:935)
E/AndroidRuntime( 5263): 	at android.os.Handler.handleCallback(Handler.java:873)
E/AndroidRuntime( 5263): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime( 5263): 	at android.os.Looper.loop(Looper.java:193)
E/AndroidRuntime( 5263): 	at android.app.ActivityThread.main(ActivityThread.java:6669)
E/AndroidRuntime( 5263): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 5263): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
E/AndroidRuntime( 5263): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
E/AndroidRuntime( 5263): Caused by: java.lang.RuntimeException: Font asset not found font_icons.ttf
E/AndroidRuntime( 5263): 	at android.graphics.Typeface.createFromAsset(Typeface.java:839)
E/AndroidRuntime( 5263): 	at com.instabug.library.view.IconView.<init>(IconView.java:51)
E/AndroidRuntime( 5263): 	at com.instabug.library.view.IconView.<init>(IconView.java:39)
E/AndroidRuntime( 5263): 	... 90 more
### Instabug integration code
*Make sure to remove your app token.*

### Android Version
9.0

### Device Model
emulator

**Comment by mzelzoghbi** [comment]
Hi @dimmetrius , Thanks for reporting this, Did you tried to enable MultiDex ?

**Comment by dimmetrius** [comment]
Hi @mzelzoghbi , yes

android/app/build.gradle

```gradle
    defaultConfig {
        ...
        minSdkVersion 21
        targetSdkVersion 28
        multiDexEnabled true
        ...
    }
```

**Comment by mzelzoghbi** [comment]
@dimmetrius  Can you share with us a sample project and the emulator version? this will help us to identify the issue and fix it.

**Comment by dimmetrius** [comment]
I just created a new flutter application and InstaBug opens without any problems. Do you have any idea what might be wrong with my current application?

**Comment by mzelzoghbi** [comment]
These kind of issues are always **Multi-dex** issues, so you should make sure that you override `attachBaseContext` method in your **Application** class and add `Multidex.install(this);` in its implementation. you should also make sure that you add the Multi-dex dependency to your **build.gradle** file
```
    implementation 'com.android.support:multidex:1.0.3'
```

@dimmetrius Please try this and let me know if this solved the issue.

**Comment by mzelzoghbi** [comment]
@dimmetrius Can you also try updating the SDK version to **8.4.0**

**Comment by dimmetrius** [comment]
I've added:

**build.gradle**
```gradle
    implementation 'com.android.support:multidex:1.0.3'
    implementation 'com.instabug.library:instabug:8.4.0'
```

**CustomFlutterApplication**

```java
public class CustomFlutterApplication extends FlutterApplication {

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        MultiDex.install(this);
    }

    @Override
    public void onCreate() {
        super.onCreate();

        new Instabug.Builder(CustomFlutterApplication.this, "XXX")
                .setInvocationEvents(InstabugInvocationEvent.SHAKE, InstabugInvocationEvent.SCREENSHOT)
                .build();
        Instabug.setWelcomeMessageState(WelcomeMessage.State.DISABLED);

        CrashReporting.setState(Feature.State.DISABLED);
        BugReporting.setReportTypes(BugReporting.ReportType.BUG, BugReporting.ReportType.FEEDBACK);

    }
}
```

but the app still crashes :(

**Comment by mzelzoghbi** [comment]
@dimmetrius we are sorry for this inconvenience, Can you please share with us the following info?
- Android build tools version
- Android Studio Version
- Support libraries version
- min / target / compile SDK Versions
- Emulator/device that you are testing on version/model
- Is R8 enabled?
- Is Jetifier enabled?
- The dependencies that you are using in your **build.gradle** file

This will really help us to reproduce the issue locally and solve the issue

**Comment by dimmetrius** [comment]
Could it be related to fonts?


Caused by java.lang.RuntimeException
Font asset not found font_icons.ttf
...
com.instabug.library.view.IconView.<init> (IconView.java:51)
com.instabug.library.view.IconView.<init> (IconView.java:39)
...
com.instabug.bug.view.a.onCreateViewHolder (AttachmentsAdapter.java:82)

**Comment by mzelzoghbi** [comment]
It may be, but we still don't know why this happened in your project and can't be reproduced on a sample project. So we are trying to know all the configurations that you have done in your project and figuring out what is happening wrong.

**Comment by dimmetrius** [comment]
- Android build tools version : **buildToolsVersion '28.0.3'**
- Android Studio Version: **Android Studio 3.3
Build #AI-182.5107.16.33.5199772, built on December 25, 2018**
- min / target / compile SDK Versions:
**compileSdkVersion 28
minSdkVersion 21
targetSdkVersion 28**
- Emulator/device that you are testing on version/model:
<img width="313" alt="avd" src="https://user-images.githubusercontent.com/2039160/59438738-bcc7da80-8dfc-11e9-8782-c2e3840a68b0.png">

- Is R8 enabled: **NO**
- Is Jetifier enabled:**android.enableJetifier=true android.useAndroidX=true**
- The dependencies that you are using in your build.gradle file

```gradle
dependencies {
    // compile fileTree(include: ['*.jar'], dir: 'libs')
    implementation 'com.android.support:multidex:1.0.3'
    implementation "com.android.support:support-v4:28.0.0"
    implementation 'com.android.support:appcompat-v7:28.0.0'
    implementation 'androidx.legacy:legacy-support-v13:1.0.0'
    implementation 'androidx.appcompat:appcompat:1.1.0-alpha03'
    implementation 'androidx.annotation:annotation:1.1.0-alpha02'
    implementation 'com.google.firebase:firebase-core:16.0.8'
    implementation 'com.google.firebase:firebase-auth:16.2.0'
    implementation 'com.google.firebase:firebase-storage:16.1.0'
    implementation 'com.crashlytics.sdk.android:crashlytics:2.9.9'
    implementation 'com.instabug.library:instabug:8.4.0'
    androidTestImplementation 'androidx.annotation:annotation:1.1.0-alpha02'
    androidTestImplementation 'androidx.test:runner:1.1.2-alpha02'
    androidTestImplementation 'androidx.test:rules:1.1.2-alpha02'
}
```

**Comment by mzelzoghbi** [comment]
Hey @dimmetrius 

We created a snapshot that should fix the issue, Can you please try it and let us know if everything works out?

```
implementation 'com.instabug.library:instabug:8.4.0.1-SNAPSHOT'
```
And don't forget to add this to your app **build.gradle** before your dependencies
```
repositories {
    ....
    maven {
        url "https://oss.sonatype.org/content/repositories/snapshots"
    }
    ...
}
```

**Comment by dimmetrius** [comment]
@mzelzoghbi Great. It works! 👍
when I can use the fix in production?

**Comment by mzelzoghbi** [comment]
Hey @dimmetrius, This fix was included in our latest release **8.5.0** so I'll close this issue. Please let us know if you have any issues.
